tests/integration_test/stats_test.go

```diff
@@ -321,6 +321,46 @@ func TestStatsEndpoints(t *testing.T) {
 	assert.Equal(t, int64(90), overallStats.ActiveTvl)
 }
 
+func TestReturnEmptyArrayWhenNoStakerStatsFound(t *testing.T) {
+	testServer := setupTestServer(t, nil)
+	defer testServer.Close()
+	stakerPk, err := testutils.RandomPk()
+	require.NoError(t, err)
+	stakerStats := fetchStakerStatsEndpoint(t, testServer, stakerPk)
+	assert.Equal(t, 0, len(stakerStats))
+}
+
+func FuzzReturnStakerStatsByStakerPk(f *testing.F) {
+	attachRandomSeedsToFuzzer(f, 3)
+	f.Fuzz(func(t *testing.T, seed int64) {
+		testServer := setupTestServer(t, nil)
+		defer testServer.Close()
+		r := rand.New(rand.NewSource(seed))
+		events := testutils.GenerateRandomActiveStakingEvents(r, &testutils.TestActiveEventGeneratorOpts{
+			NumOfEvents:        testutils.RandomPositiveInt(r, 10),
+			Stakers:            testutils.GeneratePks(10),
+			EnforceNotOverflow: true,
+		})
+		sendTestMessage(testServer.Queues.ActiveStakingQueueClient, events)
+		time.Sleep(10 * time.Second)
+
+		// Find the unique staker pks
+		var stakerPks []string
+		for _, e := range events {
+			// append into stakerPks if it's not already there
+			if !testutils.Contains(stakerPks, e.StakerPkHex) {
+				stakerPks = append(stakerPks, e.StakerPkHex)
```

Comment:
nit: seems we can use a map to contain events so that we can assert the stats content for each staker later